### ISSUE TITLE
File name and location is always "go-hclog@v0.15.0/stdlog.go:45"

### ISSUE THREAD

**Issue by vsekhar** [hashicorp/go-hclog#82]
When [using hclog with the system-wide logger](https://github.com/hashicorp/go-hclog#use-this-with-code-that-uses-the-standard-library-logger), the file name and location always refers to a file in the hclog library.

I suspect the call depth must be manipulated to account for the additional indirection, as hinted to in [the documentation for the standard logger Output function](https://golang.org/pkg/log/#Logger.Output).

**Comment by vsekhar** [comment]
Probably need to add a check for "stdlog.go" here:

https://github.com/hashicorp/go-hclog/blob/e59fd7e1110848012d4335fcd0d3bb88e0cdf9e7/intlogger.go#L203

**Comment by vsekhar** [comment]
PR #77 is related to this.

Post PR #77, entries originated from a `stdlogAdapter` probably need a call-depth of 6 rather than 3 to account for the frames of the standard log package on the stack.

**Comment by dnephin** [comment]
Thank you for reporting this issue! I looked into it as part of #77. It seems there are two ways to get the line file and number in the log message. One is the stdlib logger using `log.Llongfile` or `log.shortfile` flags, and the other is setting the `IncludeLocation` to true.

I wrote a test that shows that `FromStandardLogger` works correct with `IncludeLocation`, but the stdlib flags will not work because the caller location is different (as you pointed out).

`FromStandardLogger` could call `log.Output` to change the callstack depth, but that seems like unexpected behaviour. If other functions are using the stdlogger directly they would end up with the wrong frame.

I think the correct workaround would be either to not use the stdlib flags and rely on `go-hclog` to set the filename, or to pass in a logger that has already set `Output(6, "")` before passing the logger to `FromStandardLogger`. Maybe we need to document this in the docstring for `FromStandardLogger`.
